src/include/byteorder.h

```diff
@@ -16,6 +16,10 @@ struct ceph_le {
 private:
   T v;
 public:
+  ceph_le() = default;
```

Comment:
will replace `ceph_le` as a template alias of `endian_arithmetic`, once we have the luxury of using boost with the https://github.com/boostorg/endian/issues/28 fix.